docs/overview.rst

```diff
@@ -1,380 +1,306 @@
+- :meth:`web3.eth.getTransactionCount() <web3.eth.Eth.getTransactionCount>`
+- :meth:`web3.eth.getUncleByBlock() <web3.eth.Eth.getUncleByBlock>`
+- :meth:`web3.eth.getUncleCount() <web3.eth.Eth.getUncleCount>`
 
-        >>> Web3.toChecksumAddress('0xd3cda913deb6f67967b99d67acdfa1712c293601')
-        '0xd3CdA913deB6f67967B99D67aCDFa1712C293601'
 
-.. _EIP55: https://github.com/ethereum/EIPs/issues/55
+Making Transactions
+-------------------
 
+The most common use cases will be satisfied with
+:meth:`sendTransaction <web3.eth.Eth.sendTransaction>` or the combination of
+:meth:`signTransaction <web3.eth.Eth.signTransaction>` and
+:meth:`sendRawTransaction <web3.eth.Eth.sendRawTransaction>`. If interacting
+with a smart contract, see the next section, :ref:`Contracts <overview_contracts>`.
 
-.. _overview_hashing:
 
-Cryptographic Hashing
-~~~~~~~~~~~~~~~~~~~~~
+API
+^^^
 
-.. py:classmethod:: Web3.keccak(primitive=None, hexstr=None, text=None)
+- :meth:`web3.eth.sendTransaction() <web3.eth.Eth.sendTransaction>`
+- :meth:`web3.eth.signTransaction() <web3.eth.Eth.signTransaction>`
+- :meth:`web3.eth.sendRawTransaction() <web3.eth.Eth.sendRawTransaction>`
+- :meth:`web3.eth.replaceTransaction() <web3.eth.Eth.replaceTransaction>`
+- :meth:`web3.eth.modifyTransaction() <web3.eth.Eth.modifyTransaction>`
+- :meth:`web3.eth.waitForTransactionReceipt() <web3.eth.Eth.waitForTransactionReceipt>`
+- :meth:`web3.eth.getTransactionReceipt() <web3.eth.Eth.getTransactionReceipt>`
+- :meth:`web3.eth.sign() <web3.eth.Eth.sign>`
+- :meth:`web3.eth.signTypedData() <web3.eth.Eth.signTypedData>`
+- :meth:`web3.eth.estimateGas() <web3.eth.Eth.estimateGas>`
+- :meth:`web3.eth.generateGasPrice() <web3.eth.Eth.generateGasPrice>`
+- :meth:`web3.eth.setGasPriceStrategy() <web3.eth.Eth.setGasPriceStrategy>`
 
-    Returns the Keccak-256 of the given value. Text is encoded to UTF-8 before
-    computing the hash, just like Solidity. Any of the following are
-    valid and equivalent:
 
-    .. code-block:: python
+.. _overview_contracts:
 
-        >>> Web3.keccak(0x747874)
-        >>> Web3.keccak(b'\x74\x78\x74')
-        >>> Web3.keccak(hexstr='0x747874')
-        >>> Web3.keccak(hexstr='747874')
-        >>> Web3.keccak(text='txt')
-        HexBytes('0xd7278090a36507640ea6b7a0034b69b0d240766fa3f98e3722be93c613b29d2e')
-
-.. py:classmethod:: Web3.solidityKeccak(abi_types, value)
+Contracts
+---------
 
-    Returns the Keccak-256 as it would be computed by the solidity ``keccak``
-    function on the provided ``value`` and ``abi_types``.  The ``abi_types``
-    value should be a list of solidity type strings which correspond to each
-    of the provided values.
+The two most common use cases involving smart contracts are deploying and executing
+functions on a deployed contract.
 
+Deployment requires that the contract already be compiled, with its bytecode and ABI
+available. This compilation step is typically done within
```

Comment:
Maybe take out the `typically` and reword with language like *"can be done with ..."*